### DISCUSSION TITLE
How to interpolate f(x,y,z) to f(x,z,y) (I think)

### DISCUSSION THREAD

**Discussion by ethrelfall** [firedrakeproject/firedrake#3705] [Firedrake support] [answered]
Dear community,

I think I want to be able to interpolate a function defined on mesh "a" onto mesh "b" but I want to change the coordinates around.

A simple example would be f(x,y) on "a" to f(y,x) on "b".

I want to do this because I want to use the vdegree=0 projection to perform integration but my extruded direction is "z" whereas I want to integrate over my function's "y" values.

More specifically : I've got f(x,v) on the 2D base mesh.  I want to compute \int dv' f(x,v') K(v,v') (= g(x,v)).  I don't think I can do this straight off because v' corresponds to the y-direction and not the extruded direction (which I think will always be z).

Thanks in advance!

**Comment by dham** [comment, marked as answer]
Hi Ed,

You can get an extruded mesh in a different coordinate direction by modifying its coordinates. E.g. if m is a mesh:

```
x = SpatialCoordinate(m)
m.coordinate.interpolate(as_vector((x[1], x[0])))
```
m will now be flipped over.

**Reply by ethrelfall** [reply]
Works great (I had to do m.coordinates.interpolate(as_vector((x[1], x[0])))).  Much appreciated.
